airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/excel_parser.py

```diff
@@ -0,0 +1,177 @@
+    ) -> Iterable[Dict[str, Any]]:
+        """
+        Parses records from an Excel file based on the provided configuration.
+
+        Args:
+            config (FileBasedStreamConfig): Configuration for the file-based stream.
+            file (RemoteFile): The remote file to be read.
+            stream_reader (AbstractFileBasedStreamReader): Reader to read the file.
+            logger (logging.Logger): Logger for logging information and errors.
+            discovered_schema (Optional[Mapping[str, SchemaType]]): Discovered schema for validation.
+
+        Yields:
+            Iterable[Dict[str, Any]]: Parsed records from the Excel file.
+        """
+
+        def validate_format(excel_format: Any) -> None:
+            """
+            Validates if the given format is of type ExcelFormat.
+
+            Args:
+                excel_format (Any): The format to be validated.
+
+            Raises:
+                ConfigValidationError: If the format is not ExcelFormat.
+            """
+            if not isinstance(excel_format, ExcelFormat):
+                logger.info(f"Expected ExcelFormat, got {excel_format}")
+                raise ConfigValidationError(FileBasedSourceError.CONFIG_VALIDATION_ERROR)
+
+        def open_and_parse_file(fp: Any) -> pd.DataFrame:
+            """
+            Opens and parses the Excel file.
+
+            Args:
+                fp: File pointer to the Excel file.
+
+            Returns:
+                pd.DataFrame: Parsed data from the Excel file.
+            """
+            return pd.ExcelFile(fp, engine="calamine").parse()
+
+        # Validate the format of the config
+        validate_format(config.format)
+
+        try:
+            # Open and parse the file using the stream reader
+            with stream_reader.open_file(file, self.file_read_mode, self.ENCODING, logger) as fp:
+                df = open_and_parse_file(fp)
+                # Yield records as dictionaries
+                yield from df.to_dict(orient="records")
+
+        except Exception as exc:
+            # Raise a RecordParseError if any exception occurs during parsing
+            raise RecordParseError(FileBasedSourceError.ERROR_PARSING_RECORD, filename=file.uri) from exc
+
+    @staticmethod
+    def _extract_partitions(filepath: str) -> List[str]:
+        """
+        Extracts partitions from the file path.
+
+        Args:
+            filepath (str): The path of the file.
+
+        Returns:
+            List[str]: List of extracted partitions.
+        """
+        return [unquote(partition) for partition in filepath.split(os.sep) if "=" in partition]
+
+    @property
+    def file_read_mode(self) -> FileReadMode:
+        """
+        Returns the file read mode for the Excel file.
+
+        Returns:
+            FileReadMode: The file read mode (binary).
+        """
+        return FileReadMode.READ_BINARY
+
+    @staticmethod
+    def dtype_to_json_type(current_type: Optional[str], dtype: Any) -> str:
```

Comment:
There seems to be more dtypes defined in [this table](https://pandas.pydata.org/pandas-docs/stable/user_guide/basics.html#dtypes) like other numbers like int/float and other types like `Interval`. Is it that the calamine engine does not support these?

Reply:
No, based on the source code, only these formats are supported - **https://github.com/dimastbk/python-calamine/blob/63b671eee7c9f8738c17c924a941ab5361abc567/src/types/cell.rs#L8**